test_common/harness/imageHelpers.cpp

```diff
@@ -484,6 +484,28 @@ size_t compare_scanlines(const image_descriptor *imageInfo, const char *aPtr,
             }
             break;
 
+            case CL_SNORM_INT8: {
+                cl_uchar aPixel = *(cl_uchar *)aPtr;
+                cl_uchar bPixel = *(cl_uchar *)bPtr;
+                // -1.0 is defined as 0x80 and 0x81
+                if ((aPixel != bPixel) && ((aPixel | bPixel) != 0x81))
```

Comment:
This might be a bit too clever, since as-written it will pass the test if `aPixel` is `0x81` and `bPixel` is `0`, for example.

Maybe it'd be better to just pick one of the `-1.0` representations instead, something like:

```suggestion
                // -1.0 is defined as 0x80 and 0x81
                aPixel = (aPixel == 0x80) ? 0x81 : aPixel;
                bPixel = (bPixel == 0x80) ? 0x81 : bPixel;
                if (aPixel != bPixel)
```